utility/utils.py

```diff
@@ -735,3 +740,185 @@ def is_cluster_running(cluster_path):
     return config.RUN['cli_params'].get('cluster_path') and OCP.set_kubeconfig(
         os.path.join(cluster_path, config.RUN.get('kubeconfig_location'))
     )
+
+
+def decompose_html_attributes(soup, attributes):
+    """
+    Decomposes the given html attributes
+
+    Args:
+        soup (obj): BeautifulSoup object
+        attributes (list): attributes to decompose
+
+    Returns: None
+
+    """
+    for attribute in attributes:
+        tg = soup.find_all(attrs={"class": attribute})
+        for each in tg:
+            each.decompose()
+
+
+def parse_html_for_email(soup):
+    """
+    Parses the html and filters out the unnecessary data/tags/attributes
+    for email reporting
+
+    Args:
+        soup (obj): BeautifulSoup object
+
+    """
+    decompose_html_attributes(soup, ["extra", "col-links"])
+    soup.find(id="not-found-message").decompose()
+
+    for tr in soup.find_all('tr'):
+        for th in tr.find_all('th'):
+            if "Links" in th.text:
+                th.decompose()
+
+    for p in soup.find_all('p'):
+        if "(Un)check the boxes to filter the results." in p.text:
+            p.decompose()
+        if "pytest-html" in p.text:
+            data = p.text.split("by")[0]
+            p.string = data
+
+    for ip in soup.find_all('input'):
+        if not ip.has_attr('disabled'):
+            ip['disabled'] = 'true'
+
+    for td in soup.find_all('td'):
+        if "pytest" in td.text or "html" in td.text:
+            data = td.text.replace('&apos', '')
+            td.string = data
+
+    main_header = soup.find('h1')
+    main_header.string.replace_with('OCS-CI RESULTS')
+
+
+def email_reports():
+    """
+    Email results of test run
+
+    """
+    mailids = config.ENV_DATA['email']
+    recipients = []
+    [recipients.append(mailid) for mailid in mailids.split(",")]
+    sender = "ocs-ci@redhat.com"
+    msg = MIMEMultipart('alternative')
+    msg['Subject'] = f"ocs-ci results for RUN ID: {config.RUN['run_id']}"
+    msg['From'] = sender
+    msg['To'] = ", ".join(recipients)
+
+    html = config.ENV_DATA['report']
+    html_data = open(html).read()
+    soup = BeautifulSoup(html_data, "html.parser")
+
+    parse_html_for_email(soup)
+    part1 = MIMEText(soup, 'html')
+    msg.attach(part1)
+    try:
+        s = smtplib.SMTP('localhost')
+        s.sendmail(sender, recipients, msg.as_string())
+        s.quit()
+        log.info(f"Results have been emailed to {recipients}")
+    except Exception as e:
+        log.exception(e)
+
+
+def get_cluster_version_info():
```

Comment:
I have a separate pull request for version reporting opened here: https://github.com/red-hat-storage/ocs-ci/pull/185/files (the plan is to integrate it with logging and setup/teardown, and extend reporting to cover rpm versions of selected components such as rook and ceph, and other storage namespaces).

I suggest to keep your code here, and I will change it later to consume the some of the versions my code will report later, assuming your PR will be merged first.

Would it work for you?